### PR TITLE
Move tree from Cell<> to RwLock<> which is Sync

### PR DESCRIPTION
**Pull request by hansl**

### CODE REVIEW

File: src/merk/mod.rs [hunk 1]
```diff
@@ -609,17 +610,17 @@ mod test {
             let mut merk = Merk::open(&path).unwrap();
             let batch = make_batch_seq(1..10_000);
             merk.apply(batch.as_slice(), &[]).unwrap();
-            let mut tree = merk.tree.take().unwrap();
-            let walker = RefWalker::new(&mut tree, merk.source());
+            let mut tree =  merk.tree.write().unwrap();
```
Comment (fmorency) [inline src/merk/mod.rs]: nit: space

File: src/merk/mod.rs [hunk 2]
```diff
@@ -609,17 +610,17 @@ mod test {
...
 
             let mut nodes = vec![];
             collect(walker, &mut nodes);
             nodes
         };
 
         let merk = TempMerk::open(&path).unwrap();
-        let mut tree = merk.tree.take().unwrap();
-        let walker = RefWalker::new(&mut tree, merk.source());
+        let mut tree =  merk.tree.write().unwrap();
```
Comment (fmorency) [inline src/merk/mod.rs]: nit: space

**Review by fmorency — APPROVED** [review]
Lgtm!

Worth checking for lock poisoning?

**Comment by hansl** [comment]
For the record, we used to use `fmerk` (from https://github.com/FindoraNetwork/fmerk/) instead of Nomic's merk crate, but `fmerk` was not maintained so we forked Nomic's merk and this PR contains the minimum amount of work necessary to bring `merk` to what we needed to use it locally.

Pending a PR that moves this stuff in `merk` (which would be a big breaking change for them), we will likely keep using our fork here.

**Pull request merged**